crates/solidity/inputs/language/src/definition.rs

```diff
@@ -6961,7 +6981,7 @@ codegen_language_macros::compile!(Language(
             functions = []
         ),
         BuiltInType(
-            name = "$typeContractType",
+            name = "$ContractTypeType",
```

Comment:
I found a few `TypeType`. Is this expected?

Reply:
Yes, these are the types returned by the `type()` expression, hence the double `Type`. But I see a conflict here, because we also have `UserTypeType` which is the type of a user-defined value type. I'm thinking I'll rename that one to `UDVTType` or similar.

Reply:
I renamed `UserTypeType` to `UserDefinedValueType`.

Reply:
Sounds good. I also see three remaining `IntTypeType`, `ContractTypeType`, and `InterfaceTypeType`.

Reply:
Yes, those remaining are the types returned by `type()` expressions. I renamed `UserTypeType` because that wasn't something returned by `type()`.